bgpd/bgp_backend_functions.c

```diff
@@ -5165,16 +5166,15 @@ daemon_bgp_clear (struct bgp *bgp,  afi_t afi, safi_t safi,
             return CMD_WARNING;
         }
 
-        peer_deactivate(peer, AFI_IP6, SAFI_UNICAST);
```

Comment:
Why?

1. This function isn't used in original Quagga for realisation of "clear bgp". Deactivate and activate are used for other purposes there.
2. These functions lead to reset of sessions(deactivate and activate send NOTIFICATION). Both of commands 'clear soft" and "clear hard" work as "hard" in this case.  We need such functionality for "clear hard" only and we have  branch 
if (stype == BGP_CLEAR_SOFT_NONE)
          ret = peer_clear(peer);
for these purpose.
3. peer_deacivate(..) makes reset of afi-related settings. That's why these commands didn't work for AFI_IP6. For more info see peer_af_flag_reset(..) (ops-quagga/bgpd/bgpd.c)